src/libraries/Common/src/Interop/Linux/OpenLdap/Interop.Ldap.cs

```diff
@@ -67,6 +69,28 @@ internal static partial class Ldap
     {
         static Ldap()
         {
+            Assembly currentAssembly = typeof(Ldap).Assembly;
+
+            // Register callback that tries to load other libraries when the default library "libldap-2.5.so.0" not found
+            AssemblyLoadContext.GetLoadContext(currentAssembly).ResolvingUnmanagedDll += (assembly, ldapName) =>
+            {
+                IntPtr handle = IntPtr.Zero;
+
+                if (assembly != currentAssembly || ldapName != Libraries.OpenLdap)
+                {
+                    return handle;
+                }
+
+                // Try loading previous (libldap-2.4.so.2) and next (libldap-2.6.so.0) versions
+                if (NativeLibrary.TryLoad("libldap-2.4.so.2", out handle) ||
+                    NativeLibrary.TryLoad("libldap-2.6.so.0", out handle))
+                {
+                    return handle;
+                }
+
+                throw new DllNotFoundException(SR.Format(SR.LDAP_LIBRARY_NOT_FOUND, ldapName));
```

Comment:
This should just return `IntPtr.Zero`. The runtime will handle throwing `DllNotFoundException` with the library name.

Reply:
Thank you, fixed